tests/unit/test_distro.py

```diff
@@ -346,6 +346,78 @@ def test_configure_env_endpoint(self, mocker):
         assert os.environ[OTEL_LOGS_EXPORTER] == "otlp"
         assert os.environ.get("OTEL_SEMCONV_STABILITY_OPT_IN") == "http"
 
+    def test_configure_env_service_key_only(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-01.cloud.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "apm.collector.na-02.cloud.solarwinds.com",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-02.cloud.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only_dev(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "apm.collector.na-01.dev-ssp.solarwinds.com",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-01.dev-ssp.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only_stg(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "apm.collector.na-02.st-ssp.solarwinds.com",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-02.st-ssp.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only_invalid(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "www.google.com",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-01.cloud.solarwinds.com:443"
+
```

Comment:
Weird ask because it's repetitive but it'll help with spec clarity:

Could we have an additional unit test where `"SW_APM_COLLECTOR": "otel.collector.na-02.st-ssp.solarwinds.com"` then the assert is the endpoint is the default `"https://otel.collector.na-01.cloud.solarwinds.com:443"`. This is to make clear that `COLLECTOR` doesn't get resolved if `otel.collector` which I _think_ is how we address that edge case.

Reply:
Sure. Done